### PR TITLE
[Search] Add Search GA 11.4.0 features

### PR DESCRIPTION
**Pull request by ShivangiReja**
For Search GA `11.4.0` release, adding below features on the top of last 11.3.0 GA release:
* AAD support including multi cloud support
* Any bug fixes since last GA release
* Troubleshooting guide

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/30325

### CODE REVIEW

**Comment by ShivangiReja** [comment]
@pallavit - [Here](https://apiview.dev/Assemblies/Review/62e7aa9f69824035a62a5dde24730826/73794df8824347ff9d4a8fe56250c0af?diffRevisionId=6d5b3098e2e2478c840dd0a2c1272806&doc=False&diffOnly=True) is the API view comparing with GA `11.3.0` API.

File: sdk/search/Azure.Search.Documents/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,85 @@
 # Release History
 
+## 11.4.0 (Unreleased)
+
+### Features Added
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/CHANGELOG.md:5]: This is a great changelog!

File: sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+## Table of Contents
+
+* [Identifying and Troubleshooting Issues by Response Code](#troubleshooting-issues-by-response-code)
+  * [HTTP 207 Errors](#207-multi-status)
+  * [HTTP 404 Errors](#404-not-found)
+  * [HTTP 429 Errors](#429-too-many-requests)
+* [Unexpected search query results](#unexpected-search-query-results)
+* [Limits imposed by the Search service](#limits-imposed-by-the-service)
+
+# Troubleshooting Issues By Response Code
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md:16]:
```suggestion
# Troubleshooting Issues By Response Code

```

File: sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+See [this page](https://docs.microsoft.com/rest/api/searchservice/http-status-codes) for the common response status codes sent by the Azure Cognitive Search service.
+
+## 207 Multi-Status
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md:19]:
```suggestion
## 207 Multi-Status

```

File: sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+This response status indicates a partial success for an indexing operation. Some documents were successfully processed, but at least one failed. Details of the failed documents are present in the individual `IndexingResult` objects within the `IndexDocumentsResult` of the `Azure.Response`. If you want the [`IndexDocumentsAsync()`](https://docs.microsoft.com/dotnet/api/azure.search.documents.searchclient.indexdocumentsasync) method call to throw an exception on any failure, set [`IndexDocumentsOptions.ThrowOnAnyError`](https://docs.microsoft.com/dotnet/api/azure.search.documents.indexdocumentsoptions.throwonanyerror) to `true`. Each failure is then recorded in a separate `RequestFailedException` and a single `AggregateException` is thrown by the method.
+
+## 403 Forbidden
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md:22]:
```suggestion
## 403 Forbidden

```

File: sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+Returned when you pass an invalid api-key. Search service uses two types of keys to control access: admin (read-write) and query (read-only). The **admin key** grants full rights to all operations, including the ability to manage the service, create and delete indexes, indexers, and data sources. The **query key** grants read-only access to indexes and documents. Ensure that the key used for an API call provides sufficient privileges for the operation. See [here](https://docs.microsoft.com/azure/search/search-security-api-keys) for details about managing API keys.
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md:23]:
```suggestion
Returned when you pass an invalid API key. Search service uses two types of keys to control access: admin (read-write) and query (read-only). The **admin key** grants full rights to all operations, including the ability to manage the service, create and delete indexes, indexers, and data sources. The **query key** grants read-only access to indexes and documents. Ensure that the key used for an API call provides sufficient privileges for the operation. See [here](https://docs.microsoft.com/azure/search/search-security-api-keys) for details about managing API keys.
```

File: sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+## 404 Not Found
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md:25]:
```suggestion
## 404 Not Found

```

File: sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+Returned when a resource does not exist on the server. If you are managing or querying an index, check the syntax and verify the index name is specified correctly.
+
+## 429 too many requests
+If this error occurs while you are trying to create an index, it means you already have the maximum number of indexes allowed for your pricing tier. A count of the indexes stored in Azure Cognitive Search is visible in the search service dashboard on the [Azure portal](https://portal.azure.com/). To view the indexes by name, click the Index tile. Alternatively, you can also get a list of the indexes by name using the [GetIndexNames() method](https://docs.microsoft.com/dotnet/api/azure.search.documents.indexes.searchindexclient.getindexnamesasync).
+
+If this error occurs during document upload, it indicates that you've exceeded your quota on the number of documents per index. You must either create a new index or upgrade for higher capacity limits.
+
+# Unexpected Search Query Results
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/TROUBLESHOOTING.md:33]:
```suggestion
# Unexpected Search Query Results

```

File: sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs [hunk 8]
```diff
@@ -50,6 +50,21 @@ public class SearchIndexClient
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchIndexClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">Required. The URI endpoint of the Search service. This is likely to be similar to "https://{search_service}.search.windows.net". The URI must use HTTPS.</param>
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs:56]:
```suggestion
        /// <param name="endpoint">Required. The URI endpoint of the Search service. This is likely to be similar to "https://{search_service}.search.windows.net". The endpoint must use HTTPS.</param>
```

File: sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs [hunk 9]
```diff
@@ -79,6 +94,35 @@ public class SearchIndexClient
             _version = options.Version;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchIndexClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">Required. The URI endpoint of the Search service. This is likely to be similar to "https://{search_service}.search.windows.net". The URI must use HTTPS.</param>
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs:100]:
```suggestion
        /// <param name="endpoint">Required. The URI endpoint of the Search service. This is likely to be similar to "https://{search_service}.search.windows.net". The endpoint must use HTTPS.</param>
```

File: sdk/search/Azure.Search.Documents/tests/DocumentOperations/SearchTests.cs [hunk 10]
```diff
@@ -1008,6 +1008,30 @@ await foreach (Page<SearchResult<Hotel>> page in response.Value.GetResultsAsync(
...
         }
 
+        [Test]
+        public void SearchOptionsCanBeCopied()
+        {
+            SearchOptions source = new();
+
+            source.Facets = new List<string> { "facet1", "facet2" };
+            source.Filter = "searchFilter";
+            source.Select = null;
```
Comment (pallavit) [inline sdk/search/Azure.Search.Documents/tests/DocumentOperations/SearchTests.cs:1018]: Should this test not be in the preview release?

File: sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexerClient.cs [hunk 11]
```diff
@@ -77,6 +92,34 @@ public class SearchIndexerClient
             _version = options.Version;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SearchIndexerClient"/> class.
+        /// </summary>
+        /// <param name="endpoint">Required. The URI endpoint of the Search service. This is likely to be similar to "https://{search_service}.search.windows.net". The URI must use HTTPS.</param>
```
Comment (jsquire) [inline sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexerClient.cs:98]:
```suggestion
        /// <param name="endpoint">Required. The URI endpoint of the Search service. This is likely to be similar to "https://{search_service}.search.windows.net". The endpoint must use HTTPS.</param>
```

**Review by jsquire — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/tests/DocumentOperations/SearchTests.cs (hunk 10 above)
Comment (ShivangiReja) [reply sdk/search/Azure.Search.Documents/tests/DocumentOperations/SearchTests.cs:1018]: No, all these properties are available in the last GA release. This test case is testing that [session-id is passed or not](https://github.com/Azure/azure-sdk-for-net/blob/6cb4b4360c6ce684435bcb2d888617267d10d8bf/sdk/search/Azure.Search.Documents/tests/DocumentOperations/SearchTests.cs#L1019).

For more details [here](https://github.com/Azure/azure-sdk-for-net/issues/27549) is an issue and [fix](https://github.com/Azure/azure-sdk-for-net/pull/27582) for this.

Comment (pallavit) [reply sdk/search/Azure.Search.Documents/tests/DocumentOperations/SearchTests.cs:1018]: Got it - thanks!

**Review by pallavit — APPROVED** [review]

**Pull request merged**